src/dumper_thread.c

```diff
@@ -399,6 +430,13 @@ cr_dumper_thread(gpointer data, gpointer user_data)
         }
     } else {
         // Just gen XML from old loaded metadata
+        if ( task->media_id ) {
+            // need chunk to store location_base foreach package
+            if ( ! md->chunk )
+                md->chunk = g_string_chunk_new (PACKAGE_CHUNK_SIZE);
```

Comment:
This isn't necessary and in fact it adds overhead because of thousands of new chunks are created, and each with size 2kB.
You should use the same approach as used for ``location_href`` in this function:
https://github.com/M0ses/createrepo_c/blob/split_option/src/dumper_thread.c#L345-L346
I.e. Change definition of ``location_base`` from ``const gchar *location_base`` into ``_cleanup_free_ gchar *location_base = NULL;``. Then you should malloc the location_base string and store it into this variable (note it will be automatically freed when program flow leaves the context of the function).
You only need to handle buffering in the same way as it's done for location_href - just use ``strdup`` for the location_base and add the copy to the buffered task:
https://github.com/M0ses/createrepo_c/blob/split_option/src/dumper_thread.c#L498-L499
Also don't forget to add a free call after this line:
https://github.com/M0ses/createrepo_c/blob/split_option/src/dumper_thread.c#L575

This has several benefits:
* You don't have to deal with GStringChunks which brings a lot of overhead if used for only one and short string.
* Handling of ``location_href`` and ``location_base`` will be consistent.
* You don't have to export ``PACKAGE_CHUNK_SIZE`` in ``package.h`` header file. (Btw. it's bad to export a constant without library prefix (``CR_``) in header file which is part of public API. Also this constant is pretty useless for external consumers and it should stay hidden)

Btw. you don't have to fix this, I can take a look on this tomorrow or maybe during Friday and do the needed refactoring.